pkg/transaction/controller.go

```diff
@@ -96,16 +100,15 @@ func (C *Controller) verifyBalance(amount float64) {
 		return
 	}
 	currentBalance, _ := balanceRPCReply["result"].(string)
-	balance, _ := big.NewInt(0).SetString(currentBalance[2:], 16)
-	balance = common.NormalizeAmount(balance)
-	transfer := big.NewInt(int64(amount * denominations.Nano))
-
-	tns := float64(transfer.Uint64()) / denominations.Nano
-	bln := float64(balance.Uint64()) / denominations.Nano
+	bal, _ := new(big.Int).SetString(currentBalance[2:], 16)
+	balance := numeric.NewDecFromBigInt(bal)
+	// Add transaction amount and gas price in Atto
+	gasAsDec := numeric.NewDecFromBigInt(big.NewInt(int64(C.transactionForRPC.params["gas-price"].(uint64))))
+	total := C.transactionForRPC.params["transfer-amount"].(numeric.Dec).Add(gasAsDec)
 
-	if tns > bln {
+	if total.GT(balance) {
```

Comment:
maybe GTE?

Reply:
Transaction total cost == balance is valid. Your balance should be exactly 0 after.